src/util.ts

```diff
@@ -1,7 +1,86 @@
 import { debug, error, info, warning } from "@actions/core";
+import moment from "moment";
 
 import { ActionLogger } from "./github/types";
+import { MonthWithMatch } from "./report/types";
 
 export function generateCoreLogger(): ActionLogger {
   return { info, debug, warn: warning, error };
 }
+
+/**
+ * Extract the matches calculated from a date
+ * @param dates An object that extends { date: string } with it's own custom value
+ * @param getAmount The method to extract such custom value from the object to add to the metrics
+ * @param returnAverage True if the result should be the average for the metrics for a month,
+ * false if it must be the accumulation
+ * @returns Returns an array of objects of type {month: "MM YY", matches: number}
+ */
+export const extractMatchesFromDate = <T extends { date: string }>(
+  dates: T[],
+  getAmount: (value: T) => number,
+  returnAverage: boolean,
+): MonthWithMatch[] => {
+  dates.sort((a, b) => (a.date > b.date ? 1 : -1));
+
+  // We get the month of the first date
+  let currentMonth = moment(dates[0].date).startOf("month");
+
+  const monthsWithMatches: MonthWithMatch[] = [];
+
+  let currentMonthValues: number[] = [];
+  for (const dateObj of dates) {
+    const amountToAdd = getAmount(dateObj);
+    // If it happened in the same month
+    if (currentMonth.diff(moment(dateObj.date), "month") == 0) {
+      currentMonthValues.push(amountToAdd);
+    } else {
+      let matches: number;
+      // If we have it set to get the average on that time
+      if (returnAverage) {
+        // We get the average of time
+        matches = calculateAverage(currentMonthValues);
+      } else {
+        // Else, we simply add the total amount
+        matches = currentMonthValues.reduce((a, b) => a + b, 0);
+      }
+      // We push the previous match and reset it
+      monthsWithMatches.push([currentMonth.format("MMM YYYY"), matches]);
```

Comment:
Looks like a mistake. Should it be `"MM YY"` like JSDOC days?

Reply:
I prefer `MMM YYYY` because it gives _Feb 2024_ vs `MM YY` which gives _02 24_

I feel like it is a bit more clear.